From 166be38cfb0b832f06c5ec8b3831a6a9cb16314e Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sun, 22 Mar 2020 17:15:38 -0400
Subject: [PATCH 2/2] Fix graphviz_wrap.c _wrap_agread for Pypy3

---
 pygraphviz/graphviz_wrap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pygraphviz/graphviz_wrap.c b/pygraphviz/graphviz_wrap.c
index adf0411..484e821 100644
--- a/pygraphviz/graphviz_wrap.c
+++ b/pygraphviz/graphviz_wrap.c
@@ -3463,7 +3463,7 @@ SWIGINTERN PyObject *_wrap_agread(PyObject *SWIGUNUSEDPARM(self), PyObject *args
     }
 #endif
     mode_obj1 = PyObject_GetAttrString(obj0, "mode");
-#if !defined(PYPY_VERSION)
+#if PY_VERSION_HEX >= 0x03000000
     mode_byte_obj1 = PyUnicode_AsUTF8String(mode_obj1);
 #else
     mode_byte_obj1 = mode_obj1;
-- 
2.21.0

