diff --git a/setup.py b/setup.py
index dbfae2d..54bbfea 100644
--- a/setup.py
+++ b/setup.py
@@ -66,16 +66,7 @@ if __name__ == "__main__":
         # gvplugin_*). The main graphviz libs (cgraph etc.) are in the
         # parent dir
         library_dirs = [
-            "/usr/lib/x86_64-linux-gnu",  # Ubuntu x86_64
-            "/usr/lib/x86_64-linux-gnu/graphviz",
-            "/opt/homebrew/lib",  # Macos, homebrew aarch64
-            "/opt/homebrew/lib/graphviz",
-            "/usr/lib64",  # Fedora
-            "/usr/lib64/graphviz",
-            "/usr/local/lib",  # source install / macos homebrew x86_64
-            "/usr/local/lib/graphviz",
-            "/opt/local/lib",
-            "/opt/local/lib/graphviz",
+            os.path.join(os.environ["PREFIX"], "lib/graphviz")
         ]
         # runtime_library_dirs must not be defined with windows else setup will fail
         extra_kwargs = {} if WINDOWS else {"runtime_library_dirs": library_dirs}
